pkg/cloud/aws/aws.go

```diff
@@ -119,12 +122,15 @@ func (a *awsProvider) PrepareSubmarinerClusterEnv() error {
 	}, a.reporter); err != nil {
 		return err
 	}
-	if err := a.cloudPrepare.PrepareForSubmariner(cpapi.PrepareForSubmarinerInput{
-		InternalPorts: []cpapi.PortSpec{
-			{Port: constants.SubmarinerRoutePort, Protocol: "udp"},
-		},
-	}, a.reporter); err != nil {
-		return err
+
+	if a.cniType != cni.OVNKubernetes {
+		if err := a.cloudPrepare.PrepareForSubmariner(cpapi.PrepareForSubmarinerInput{
+			InternalPorts: []cpapi.PortSpec{
+				{Port: constants.SubmarinerRoutePort, Protocol: "udp"},
+			},
+		}, a.reporter); err != nil {
+			return err
+		}
 	}
```

Comment:
Is the difference in behaviour here intentional? For the other providers, `PrepareForSubmariner` is called anyway, here it isn’t called at all.

Reply:
@skitt I wasn't sure which of the two is preferred here. So decided to write code this way so it comes up in review and get a better answer. All `PrepareForSubmariner` does today is add SG rules. So, do we need to call if the list is empty, or we still call it in case we add more stuff to it in `cloud` later down the line.

@sridhargaddam Thoughts?